### PR TITLE
[geometry-1] Add DOMRect test for Editor’s Draft.

### PR DESCRIPTION
**Pull request by hwanseung**
https://drafts.fxtf.org/geometry-1/#DOMRect

### CODE REVIEW

**Comment by w3c-bots** [comment]
[View the complete job log.](https://travis-ci.org/w3c/web-platform-tests/jobs/227038288)

# Lint #

Passed

**Comment by w3c-bots** [comment]
[View the complete job log.](https://travis-ci.org/w3c/web-platform-tests/jobs/227038290)

# Firefox (nightly channel) #
Testing web-platform-tests at revision 0510ab2ead92e06acd60053d5fa9181e7c22a4b7
Using browser at version BuildID 20170428100438; SourceStamp 84762dbeb5380461fe27f0afa0e27e8ba9dd3b01
Starting 10 test iterations
All results were stable
## All results ##
<details>
<summary>1 test ran</summary>
<details>
<summary><a href="http://w3c-test.org/submissions/5740/css/geometry-1/DOMRect-002.html">/css/geometry-1/DOMRect-002.html</a></summary>


|                   Subtest                    | Results |                                Messages                               |
|----------------------------------------------|---------|-----------------------------------------------------------------------|
|                                              | OK      |                                                                       |
| `testConstructor0`                           | PASS    |                                                                       |
| `testConstructor1`                           | FAIL    | `Not enough arguments to DOMRect.`                                    |
| `testConstructor2`                           | FAIL    | `Not enough arguments to DOMRect.`                                    |
| `testConstructor3`                           | FAIL    | `Not enough arguments to DOMRect.`                                    |
| `testConstructor4`                           | PASS    |                                                                       |
| `testConstructor5`                           | PASS    |                                                                       |
| `testConstructorNegativeWidth`               | PASS    |                                                                       |
| `testConstructorNegativeHeight`              | PASS    |                                                                       |
| `testConstructorNegativeWidthHeight`         | PASS    |                                                                       |
| `testConstructorUndefined1`                  | FAIL    | `assert_equals: Expected value for width is 0 expected 0 but got NaN` |
| `testConstructorUndefined2`                  | PASS    |                                                                       |
| `testConstructorString1`                     | PASS    |                                                                       |
| `testConstructorString2`                     | PASS    |                                                                       |
| `testReadOnlyConstructor0`                   | FAIL    | `Illegal constructor.`                                                |
| `testReadOnlyConstructor1`                   | FAIL    | `Illegal constructor.`                                                |
| `testReadOnlyConstructor2`                   | FAIL    | `Illegal constructor.`                                                |
| `testReadOnlyConstructor3`                   | FAIL    | `Illegal constructor.`                                                |
| `testReadOnlyConstructor4`                   | FAIL    | `Illegal constructor.`                                                |
| `testReadOnlyConstructor5`                   | FAIL    | `Illegal constructor.`                                                |
| `testReadOnlyConstructorNegativeWidth`       | FAIL    | `Illegal constructor.`                                                |
| `testReadOnlyConstructorNegativeHeight`      | FAIL    | `Illegal constructor.`                                                |
| `testReadOnlyConstructorNegativeWidthHeight` | FAIL    | `Illegal constructor.`                                                |
| `testReadOnlyConstructorUndefined1`          | FAIL    | `Illegal constructor.`                                                |
| `testReadOnlyConstructorUndefined2`          | FAIL    | `Illegal constructor.`                                                |
| `testReadOnlyConstructorString1`             | FAIL    | `Illegal constructor.`                                                |
| `testReadOnlyConstructorString2`             | FAIL    | `Illegal constructor.`                                                |
| `testSetReadOnlyAttributes`                  | PASS    |                                                                       |
| `testSetAttributes`                          | PASS    |                                                                       |
| `testReadOnlySetReadOnlyAttributes`          | FAIL    | `Illegal constructor.`                                                |
| `testReadOnlySetAttributes`                  | FAIL    | `Illegal constructor.`                                                |
| `testFromRect`                               | FAIL    | `DOMRect.fromRect is not a function`                                  |
| `testReadOnlyFromRect`                       | FAIL    | `DOMRectReadOnly.fromRect is not a function`                          |

</details>
</details>

**Comment by w3c-bots** [comment]
[View the complete job log.](https://travis-ci.org/w3c/web-platform-tests/jobs/227038296)

# Chrome (unstable channel) #
Testing web-platform-tests at revision 0510ab2ead92e06acd60053d5fa9181e7c22a4b7

**Comment by hwanseung** [comment]
@zcorpan could you reivew this patch?

File: css/geometry-1/DOMRect-002.html [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+                    { x: 1, y: 2, width: 3, height: 4, top: 2, right: 4, bottom: 6, left: 1 });
+            }, prefix + 'String1');
+            test(function() {
+                checkDOMRect(new constructor("a", "b", "c", "d"),
+                    { x: NaN, y: NaN, width: NaN, height: NaN, top: NaN, right: NaN, bottom: NaN, left: NaN });
+            }, prefix + 'String2');
+        }
+
+        function checkDOMRect(r, exp) {
+            assert_equals(r.x, exp.x, "Expected value for x is " + exp.x);
```
Comment (zcorpan) [inline css/geometry-1/DOMRect-002.html]: The message will already say what actual and expected values are. So the third argument should only say what it is checking. e.g. just "x" or "Property x"

File: css/geometry-1/DOMRect-002.html [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+                    { x: 2, y: 2, width: 4, height: -4, top: -2, right: 6, bottom: 2, left: 2 });
+            }, prefix + 'NegativeHeight');
+            test(function() {
+                checkDOMRect(new constructor(2, 2, -4, -4),
+                    { x: 2, y: 2, width: -4, height: -4, top: -2, right: 2, bottom: 2, left: -2 });
+            }, prefix + 'NegativeWidthHeight');
+            test(function() {
+                checkDOMRect(new constructor(0, 0, undefined, 4),
+                    { x: 0, y: 0, width: 0, height: 4, top: 0, right: 0, bottom: 4, left: 0 });
+            }, prefix + 'Undefined1');
```
Comment (zcorpan) [inline css/geometry-1/DOMRect-002.html]: Avoid numbers in test names. Describe what the test is doing in the test name instead.

**Comment by zcorpan** [comment]
In general I'd like test names to be better and also try to avoid duplicated code. See https://github.com/w3c/web-platform-tests/pull/5719/files for inspiration.

**Comment by w3c-bots** [comment]
These tests are now available on [w3c-test.org](https://w3c-test.org/submissions/5740/)

File: css/geometry-1/DOMRect-002.html (hunk 2 above)
Comment (hwanseung) [inline css/geometry-1/DOMRect-002.html]: done.

File: css/geometry-1/DOMRect-002.html (hunk 1 above)
Comment (hwanseung) [inline css/geometry-1/DOMRect-002.html]: done.

**Comment by hwanseung** [comment]
@zcorpan could you review this patch again?

**Comment by zcorpan** [comment]
Could you please check the "allow edits from maintainers" checkbox in the sidebar? I have 2 commits I would like to push to this branch.

**Comment by hwanseung** [comment]
i checked it. thanks.

**Comment by zcorpan** [comment]
Thanks, pushed.

**Review by zcorpan — APPROVED** [review]

**Comment by hwanseung** [comment]
@zcorpan this patch was not passed test. could you check it?

**Comment by w3c-bots** [comment]
[View the complete job log.](https://travis-ci.org/w3c/web-platform-tests/jobs/229935604)

# Firefox (nightly) #

# Firefox (nightly channel) #
Testing web-platform-tests at revision f532359d708d111e48ae51b72a05ce4d2f041e73
Using browser at version BuildID 20170507100239; SourceStamp 37a5b7f6f101df2eb292b1b6baaf1540c9920e20
Starting 10 test iterations
All results were stable
## All results ##
<details>
<summary>1 test ran</summary>
<details>
<summary><a href="http://w3c-test.org/submissions/5740/css/geometry-1/DOMRect-002.html">/css/geometry-1/DOMRect-002.html</a></summary>


|                           Subtest                            | Results |                    Messages                   |
|--------------------------------------------------------------|---------|-----------------------------------------------|
|                                                              | OK      |                                               |
| `DOMRect constructor without parameter`                      | PASS    |                                               |
| `DOMRect constructor with one parameter`                     | FAIL    | `Not enough arguments to DOMRect.`            |
| `DOMRect constructor with two parameters`                    | FAIL    | `Not enough arguments to DOMRect.`            |
| `DOMRect constructor with three parameters`                  | FAIL    | `Not enough arguments to DOMRect.`            |
| `DOMRect constructor with four parameters`                   | PASS    |                                               |
| `DOMRect constructor with five parameters`                   | PASS    |                                               |
| `DOMRect constructor with negative width`                    | PASS    |                                               |
| `DOMRect constructor with negative height`                   | PASS    |                                               |
| `DOMRect constructor with negative width and height`         | PASS    |                                               |
| `DOMRect constructor with undefined`                         | FAIL    | `assert_equals: width expected 0 but got NaN` |
| `DOMRect constructor with NaN and infinity and null`         | PASS    |                                               |
| `DOMRect constructor with number string`                     | PASS    |                                               |
| `DOMRect constructor with character string`                  | PASS    |                                               |
| `DOMRectReadOnly constructor without parameter`              | FAIL    | `Illegal constructor.`                        |
| `DOMRectReadOnly constructor with one parameter`             | FAIL    | `Illegal constructor.`                        |
| `DOMRectReadOnly constructor with two parameters`            | FAIL    | `Illegal constructor.`                        |
| `DOMRectReadOnly constructor with three parameters`          | FAIL    | `Illegal constructor.`                        |
| `DOMRectReadOnly constructor with four parameters`           | FAIL    | `Illegal constructor.`                        |
| `DOMRectReadOnly constructor with five parameters`           | FAIL    | `Illegal constructor.`                        |
| `DOMRectReadOnly constructor with negative width`            | FAIL    | `Illegal constructor.`                        |
| `DOMRectReadOnly constructor with negative height`           | FAIL    | `Illegal constructor.`                        |
| `DOMRectReadOnly constructor with negative width and height` | FAIL    | `Illegal constructor.`                        |
| `DOMRectReadOnly constructor with undefined`                 | FAIL    | `Illegal constructor.`                        |
| `DOMRectReadOnly constructor with NaN and infinity and null` | FAIL    | `Illegal constructor.`                        |
| `DOMRectReadOnly constructor with number string`             | FAIL    | `Illegal constructor.`                        |
| `DOMRectReadOnly constructor with character string`          | FAIL    | `Illegal constructor.`                        |
| `DOMRect: set top/right/bottom/left`                         | PASS    |                                               |
| `DOMRect: set x/y/width/height`                              | PASS    |                                               |
| `DOMRectReadOnly: set top/right/bottom/left`                 | FAIL    | `Illegal constructor.`                        |
| `DOMRectReadOnly: set x/y/width/height`                      | FAIL    | `Illegal constructor.`                        |
| `DOMRect.fromRect`                                           | FAIL    | `DOMRect.fromRect is not a function`          |
| `DOMRectReadOnly.fromRect`                                   | FAIL    | `DOMRectReadOnly.fromRect is not a function`  |

</details>
</details>

**Comment by w3c-bots** [comment]
[View the complete job log.](https://travis-ci.org/w3c/web-platform-tests/jobs/229935610)

# Chrome (unstable) #

# Chrome (unstable channel) #
Testing web-platform-tests at revision f532359d708d111e48ae51b72a05ce4d2f041e73
Using browser at version 60.0.3088.3 dev
Starting 10 test iterations
All results were stable
## All results ##
<details>
<summary>1 test ran</summary>
<details>
<summary><a href="http://w3c-test.org/submissions/5740/css/geometry-1/DOMRect-002.html">/css/geometry-1/DOMRect-002.html</a></summary>


|                           Subtest                            | Results |              Messages              |
|--------------------------------------------------------------|---------|------------------------------------|
|                                                              | OK      |                                    |
| `DOMRect constructor without parameter`                      | FAIL    | `constructor is not a constructor` |
| `DOMRect constructor with one parameter`                     | FAIL    | `constructor is not a constructor` |
| `DOMRect constructor with two parameters`                    | FAIL    | `constructor is not a constructor` |
| `DOMRect constructor with three parameters`                  | FAIL    | `constructor is not a constructor` |
| `DOMRect constructor with four parameters`                   | FAIL    | `constructor is not a constructor` |
| `DOMRect constructor with five parameters`                   | FAIL    | `constructor is not a constructor` |
| `DOMRect constructor with negative width`                    | FAIL    | `constructor is not a constructor` |
| `DOMRect constructor with negative height`                   | FAIL    | `constructor is not a constructor` |
| `DOMRect constructor with negative width and height`         | FAIL    | `constructor is not a constructor` |
| `DOMRect constructor with undefined`                         | FAIL    | `constructor is not a constructor` |
| `DOMRect constructor with NaN and infinity and null`         | FAIL    | `constructor is not a constructor` |
| `DOMRect constructor with number string`                     | FAIL    | `constructor is not a constructor` |
| `DOMRect constructor with character string`                  | FAIL    | `constructor is not a constructor` |
| `DOMRectReadOnly constructor without parameter`              | FAIL    | `constructor is not a constructor` |
| `DOMRectReadOnly constructor with one parameter`             | FAIL    | `constructor is not a constructor` |
| `DOMRectReadOnly constructor with two parameters`            | FAIL    | `constructor is not a constructor` |
| `DOMRectReadOnly constructor with three parameters`          | FAIL    | `constructor is not a constructor` |
| `DOMRectReadOnly constructor with four parameters`           | FAIL    | `constructor is not a constructor` |
| `DOMRectReadOnly constructor with five parameters`           | FAIL    | `constructor is not a constructor` |
| `DOMRectReadOnly constructor with negative width`            | FAIL    | `constructor is not a constructor` |
| `DOMRectReadOnly constructor with negative height`           | FAIL    | `constructor is not a constructor` |
| `DOMRectReadOnly constructor with negative width and height` | FAIL    | `constructor is not a constructor` |
| `DOMRectReadOnly constructor with undefined`                 | FAIL    | `constructor is not a constructor` |
| `DOMRectReadOnly constructor with NaN and infinity and null` | FAIL    | `constructor is not a constructor` |
| `DOMRectReadOnly constructor with number string`             | FAIL    | `constructor is not a constructor` |
| `DOMRectReadOnly constructor with character string`          | FAIL    | `constructor is not a constructor` |
| `DOMRect: set top/right/bottom/left`                         | FAIL    | `DOMRect is not defined`           |
| `DOMRect: set x/y/width/height`                              | FAIL    | `DOMRect is not defined`           |
| `DOMRectReadOnly: set top/right/bottom/left`                 | FAIL    | `DOMRectReadOnly is not defined`   |
| `DOMRectReadOnly: set x/y/width/height`                      | FAIL    | `DOMRectReadOnly is not defined`   |
| `DOMRect.fromRect`                                           | FAIL    | `DOMRect is not defined`           |
| `DOMRectReadOnly.fromRect`                                   | FAIL    | `DOMRectReadOnly is not defined`   |

</details>
</details>

**Comment by Ms2ger** [comment]
Try rebasing.

**Comment by zcorpan** [comment]
I've rebased this on latest master.

**Comment by w3c-bots** [comment]
# Build PENDING

Started: None
Finished: None





View more information about this build on:

- [WPT PR Status](http://pulls.web-platform-tests.org/build/19123)
- [TravisCI](https://travis-ci.org/w3c/web-platform-tests/builds/283125535)

**Pull request merged**